### ISSUE TITLE
Failed Attempt event

### ISSUE THREAD

**Issue by cpitt** [OptimalBits/bull#367]
While I find Bull to be a great improvement over Kue, one thing that I find myself missing from Kue is the differentiation of `failed` and `failed attempt` events. Currently my workaround is to 
```
myQueue
  .on('failed',
    (queueJob, error) => {
      const {attempts, attemptsMade} = queueJob;
      if (attempts === attemptsMade) {
        doSomethingForFailure();
      } else {
        doSomethingElseForAttempt();
      }
    }
```

I realize that having different events for `failed` and `failed attempt` would be a breaking change and would need to wait until the next minor or even major release, but in the mean time it might be worth documenting the above work around (or another if there is a better way of achieving this) for others coming from Kue who might require this functionality. 

I'd be more than happy to create a PRs if this request seems reasonable and inline with your project.

**Comment by manast** [comment]
wouldn't be possible to add an extra argument to the event callback with lets say say a boolean ``ìsFailedAttempt```telling you what you need to know? such a change would not be a breaking change, not sure if it solves your specific requirement though.

**Comment by cpitt** [comment]
Yeah that would work as well! Having just migrated from Kue I was approaching the problem from that context. 

The only real argument I could see for a separate event is that it would be somewhat cleaner in that you wouldn't have to make any conditional checks in the failed handler to determine whether it was an attempt failure or an overall failure. However, this slight advantage is subjective and IMO is not worth a breaking change.
